app/models/notification.rb

```diff
@@ -0,0 +1,55 @@
+class Notification < ActiveRecord::Base
+  belongs_to :activity, inverse_of: :notifications, dependent: :destroy
+
+  acts_as_readable on: :updated_at
+
+  def self.notify(user, course, object, options = {})
+    include_public_activity(object)
+    unless options.key?(:activity)
+      notify_only(user, course, object, options)
+      return
+    end
+    activity_and_notify(user, course, object, options)
+  end
+
+  class << self
+    private
+
+    def include_public_activity(object)
+      return if object.class.include?(PublicActivity::Common)
+      object.class.include(PublicActivity::Common)
+    end
+
+    def notify_only(user, course, object, options)
+      if options.key?(:popup)
+        activity = object.create_activity :notification, owner: course, recipient: user
```

Comment:
because, I need to store that notification information in activities table. In other words, I store data in activities table, if one activity need to send a notification as well (most of time, an activity need to send a notification too), I store its notification type in notifications table and reference activities table for (user, course, object) information